### DISCUSSION TITLE
Welcome to Developer Track Sessions at OGC Member Meetings!  - March 2021

### DISCUSSION THREAD

**Discussion by ghobona** [opengeospatial/developer-track#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other participants of Developer Track Sessions at OGC Member Meetings. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other developers in the community.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
 * **Abide by the [OGC Principles of Conduct](https://www.ogc.org/ogc/policies/conduct)**

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chris-little** [comment]
@ghobona  Thank you for setting this up. I was never able to use Gitter, as it never made it through our firewalls and security policies. 

I am Chris Little, Chair of the Environmental Data Retrieval (EDR) API Standard WG, where "Environmental" means anything with spatiotemporal coordinates! I hope attendees will try out the API, which, fingers crossed, touch wood, and any other irrational beliefs that might help, will become a full OGC standard in a few weeks. Though there will be some editorial tidying up to do.

**Reply by ghobona** [reply]
@chris-little Welcome!

**Comment by skyNacho** [comment]
I am Ignacio "Nacho" Correas, working for Skymantics and with main interests in routing and AI (sometimes I can even combine both interests!). At Skymantics we have a working alpha of a general-purpose OGC-compliant API that we tested in a few pilots and testbeds. It currently implements the draft APIs Routes and 3D GeoVolumes. Prior to my engagement with Skymantics and OGC I founded and led the business and technical development of Zentyal, an open source alternative to Windows Server, bringing Microsoft's protocols to the Linux world.

Last year I heard of pygeoapi and since then I have been very curious to learn more about it but never found the time. It would make a lot of sense to adopt it and contribute to its development instead of continuing re-inventing the wheel with our own little project.

We are going to test new API concepts during TB17 and I was planning to try pygeoapi, so this developer track couldn't be better timed.

**Comment by tomkralidis** [comment]
Hi all: I'm Tom Kralidis, working with the Meteorological Service of Canada.  I'm a longtime contributor to OGC standards and FOSS4G software (in particular Python and Go).  Currently, I'm co-chair of the OGC API - Records Standards Working Group and also contribute to OGC API - Coverages and OGC Environmental Data Retrieval API standards.

My goal is lowering the barrier to implementation of OGC standards implementation for weather/climate/water communities (WMO), and anyone else.

Thanks and kudos to OGC for setting up a Developer Track; I'm excited to work with everyone in this space.

**Comment by serich** [comment]
I'm an OGC staff member managing [Testbed-17](https://www.ogc.org/testbed17). If interoperable running code wins (paraphrasing the [IETF](https://www.ietf.org/about/participate/tao)), then this must be where all the winners hang out. Hope you don't mind me hanging out with you.

**Comment by nilshempelmann** [comment]
Joining the introduction round :-). 
New staff at OGC but dealing with OGC Innovations and Standards since a while. Basically working in context of climate data domain. You can find some work here https://github.com/bird-house

**Comment by StephanSiemen** [comment]
Hi all! Stephan Siemen from ECMWF here. Looking forward to the session today! :-)

**Comment by pvgenuchten** [comment]
hi, paul van genuchten, geocat, i'm presenting the tutorial on ogc-api records in pygeoapi with tom this afternoon. pygeoapi is an interesting lean technology to experiment with ogc api's. i'm also implementing an ogc-api records client in qgis and a server implementation in geonetwork

**Comment by bpross-52n** [comment]
Hi, I am Benjamin, chair of the OGC API - Processes Standards Working Group (SWG) and editor of the OGC API - Processes - Part 1: Core standard.

**Comment by kcski** [comment]
Hi I am active in the CDB community. I am working in the modeling and simulation community.
Have experience in python.

**Comment by aarathimuppalla** [comment]
Hi everyone. I am Aarathi, working as a scientist in National Remote Sensing Centre, Indian Space Research Organization.
Looking forward for today's developer track.

**Comment by AndrewRadburn** [comment]
Hi, I'm Andy Radburn working as a Data Scientist at Ordnance Survey. Looking forward to building something in Python.

**Comment by ShaneMill1** [comment]
Hi all, I am Shane Mill from NOAA. I contribute to the Environmental Data Retrieval SWG and am an early implementer of the EDR-API

**Comment by joanma747** [comment]
Hi, I'm Joan Masó from UAB-CREAF. I'm the editor of the Tile Matrix Set standard and the OGC API Tiles drafts with Jerome St-Louis. I will try to participate as developers in future editions of this track. Today I'll observe.

**Comment by iandruska-ibl** [comment]
Hi, I am Igor Andruska working as a software developer at [IBL Software Engineering](https://www.iblsoft.com/). I have been a member of the EDR SWG for the past year. I am also implementing an [EDR server](https://ogcie.iblsoft.com/edr).

**Reply by iandruska-ibl** [reply]
Some EDR query examples to start with are available at https://github.com/opengeospatial/ogcapi-environmental-data-retrieval/blob/master/implementations.md#ibl-software-engineering

**Comment by NazihFino** [comment]
Hi all! Nazih Fino. Global Nomad GIS Services, Looking forward to inspiring and productive sessions today!

**Comment by joanma747** [comment]
Some links to swaggerhub examples of OGC API tiles: 

https://github.com/opengeospatial/OGC-API-Tiles
https://github.com/opengeospatial/OGC-API-Maps
https://github.com/opengeospatial/2D-Tile-Matrix-Set

OpenAPI Materials
API's examples
https://app.swaggerhub.com/apis/UAB-CREAF/ogc-api-map-tiles-opf-xmp-mt-more-1-collection/1.0.0
https://app.swaggerhub.com/apis/UAB-CREAF/ogc-api-maps-opf-xmp-more-1-collection/1.0.0
...
Domains
https://app.swaggerhub.com/domains/UAB-CREAF/ogc-api-maps/1.0.0
https://app.swaggerhub.com/domains/UAB-CREAF/ogc-api-tiles/1.0.0

**Comment by ghobona** [comment]
For those interested in the pygeoapi edr-api implementation: https://docs.pygeoapi.io/en/latest/data-publishing/ogcapi-edr.html

**Reply by ShaneMill1** [reply]
The National Weather Service's implementation is located here: http://data-api-c.mdl.nws.noaa.gov/OGC-EDR-API. 

Just want to make a note that the URL will be available publicly for today only. As soon as our on premise system is back up and running, you will be able to access our implementation 24/7 at https://data-api.mdl.nws.noaa.gov/EDR-API. 

Thank you all for listening to the demo/tutorial and please reach out to me here or at shane.mill@noaa.gov if you have any questions!

**Comment by francbartoli** [comment]
I'm Francesco Bartoli from Geobeyond, part of the pygeoapi dev team. I present the tutorial about [Retrieving vector tiles from OGC API - Tiles implementations using pygeoapi](https://docs.google.com/presentation/d/13A5QfSgepiOmd4ttjr376MjRKgx2zdJaty0LPXgqkQk/edit#slide=id.gc9353dab56_0_29)

**Comment by ghobona** [comment]
When installing pygeoapi following the steps on the pygeoapi [homepage](https://pygeoapi.io), if you run into the following errors, here is how I solved the issues.

Please note that I was installing it into an Ubuntu 18.04.1 instance.

----------

When running this command `pip3 install -r requirements.txt`

I ran into error:

`ERROR: Cython.Build.cythonize not found. Cython is required to build pyproj`

The issue was solved by installing Cython

`pip3 install Cython --install-option="--no-cython-compile"`

Then had to upgrade pip

`python3 -m pip install -U pip`

-----------

I was then able to run:

`pip3 install -r requirements.txt`


However, I ran into the error:

`ImportError: No module named setuptools`

The issue was solved by running this:

`sudo python3 setup.py install`

-----------

Then when I ran, `pygeoapi generate-openapi-document -c $PYGEOAPI_CONFIG > $PYGEOAPI_OPENAPI`

However, I encountered this error:

`"ModuleNotFoundError: No module named 'xarray'"`


The issue was solved by running this command:

`pip install xarray`

-----------

Then retried `pygeoapi generate-openapi-document -c $PYGEOAPI_CONFIG > $PYGEOAPI_OPENAPI`

But ran into an error that netcdf4-python or scipy had to be installed.

The issue was solved by running these commands:

`python -m pip install --user numpy scipy matplotlib ipython jupyter pandas sympy nose`

`pip install netcdf4`

`pip install tinydb`

-----------

I was then able to complete the installation with these commands.

`pygeoapi generate-openapi-document -c $PYGEOAPI_CONFIG > $PYGEOAPI_OPENAPI`

`pygeoapi serve`

I now have a running instance of pygeoapi!

![DEVTRK_24_03_2021_18_19_19](https://user-images.githubusercontent.com/18121038/112365560-b6ec7200-8ccf-11eb-8d68-363fe8393afe.png)

**Reply by francbartoli** [reply]
Great @ghobona! For the different providers like `xarray` for example there is a convenient `requirements-provider.txt` file to run

`pip install -r requirements-provider.txt`

in one shot.

**Comment by ghobona** [comment]
All, 

Anyone planning to do a demo should register their demo by 17:30 EDT (21:30 GMT) at https://github.com/opengeospatial/developer-track/blob/main/demonstrations.adoc

Simply create a Pull Request and we will merge it into the document.

**Reply by ghobona** [reply]
10 minute extension for registration of demos. 

Registration will close at 17:40 EDT.

**Comment by ghobona** [comment]
Until 17:15 EDT on March 25th 2021, you can vote for your favorite demo at https://twitter.com/opengeospatial
